src/SearchResults.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import HeadingRow from "./HeadingRow";
+import BodyRow from "./BodyRow";
+import CustomerProfile from "./CustomerProfile";
+import moment from "moment";
+
+const SearchResults = ({ results, setBooking }) => {
+  const [customerId, setCustomerId] = useState(null);
+  const [isSorted, setIsSorted] = useState({
+    title: false,
+    firstName: false,
+    surname: false,
+    email: false,
+    roomId: false,
+    checkInDate: false,
+    checkOutDate: false,
+    numOfNight: false
+  });
+
+  const sortBookings = event => {
+    let sortedBookings = [];
+    let reversedBookings = [];
+    let key = event.target.title;
+
+    console.log(isSorted[key]);
+    if (!isSorted[key]) {
+      if (key == "roomId") {
+        sortedBookings = results.sort((a, b) => a[key] - b[key]);
+      } else if (key == "numOfNight") {
+        sortedBookings = results.sort(
+          (a, b) =>
+            moment(a.checkOutDate).diff(a.checkInDate, "days") -
+            moment(b.checkOutDate).diff(b.checkInDate, "days")
+        );
+      } else
+        sortedBookings = results.sort((a, b) => a[key].localeCompare(b[key]));
+      const updatedkeystatus = {
+        ...isSorted,
+        [key]: true
+      };
+
+      setIsSorted(updatedkeystatus);
+      console.log(updatedkeystatus);
+      setBooking([...sortedBookings]);
+    } else {
+      if (key == "roomId") {
```

Comment:
I think that sorting logic from line 46 to 55 could be extracted to a separate function that takes the `key` as argument and returns a sorted array. `sortBookings` function is quite long and is responsible for many things so keeping the clean code principles in mind - it would be could to break down the responsibilities into smaller functions to improve readability.